tests/common/util.rs

```diff
@@ -1158,19 +1159,67 @@ pub struct TestScenario {
     pub bin_path: PathBuf,
     pub util_name: String,
     pub fixtures: AtPath,
+    log_guard: LogPrintGuard,
     tmpd: Rc<TempDir>,
 }
 
+struct LogPrintGuard {
+    log_file: PathBuf,
+}
+
+// In the context of test execution, writing to std::io::stderr() is
+// apparently handled differently than writing using eprint!-macro
+// To still be able to use the std::io::copy, this PrintSink is needed.
+struct PrintSink {}
+
+impl std::io::Write for PrintSink {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        eprint!("{}", String::from_utf8_lossy(buf));
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl Drop for LogPrintGuard {
+    fn drop(&mut self) {
+        let Ok(f) = File::open(&self.log_file) else {
+            return;
+        };
+        eprintln!("=============== collected logs: ===============");
```

Comment:
Wasn't the entire point to avoid writing to stderr altogether?

Reply:
> Wasn't the entire point to avoid writing to stderr altogether?

Please take your time to understand the usecase.
This print statement here is part of the test framework. So it will print to stderr of the test and not the stderr of the util under test.